go-controller/pkg/cni/ovs_unit_test.go

```diff
@@ -0,0 +1,406 @@
+				for _, arg := range tc.onRetArgsCmdList.OnCallMethodArgType {
+					mockCall.Arguments = append(mockCall.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range tc.onRetArgsCmdList.RetArgList {
+					mockCall.ReturnArguments = append(mockCall.ReturnArguments, ret)
+				}
+				mockCall.Once()
+			}
+
+			runner = tc.runnerInstance
+
+			e := ovsDestroy("table", "record")
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			}
+
+			mockCmd.AssertExpectations(t)
+			mockKexecIface.AssertExpectations(t)
+		})
+	}
+}
+
+func TestOvsSet(t *testing.T) {
+	mockKexecIface := new(mock_k8s_io_utils_exec.Interface)
+	mockCmd := new(mock_k8s_io_utils_exec.Cmd)
+
+	tests := []struct {
+		desc                string
+		expectedErr         error
+		onRetArgsKexecIface *ovntest.TestifyMockHelper
+		onRetArgsCmdList    *ovntest.TestifyMockHelper
+		runnerInstance      kexec.Interface
+	}{
+		{
+			desc:                "Positive test codepath for ovsSet",
+			expectedErr:         nil,
+			onRetArgsKexecIface: &ovntest.TestifyMockHelper{"Command", []string{"string", "string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
+			onRetArgsCmdList:    &ovntest.TestifyMockHelper{"CombinedOutput", []string{}, []interface{}{nil, nil}},
+			runnerInstance:      mockKexecIface,
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			if tc.onRetArgsKexecIface != nil {
+				ifaceCall := mockKexecIface.On(tc.onRetArgsKexecIface.OnCallMethodName)
+				for _, arg := range tc.onRetArgsKexecIface.OnCallMethodArgType {
+					ifaceCall.Arguments = append(ifaceCall.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range tc.onRetArgsKexecIface.RetArgList {
+					ifaceCall.ReturnArguments = append(ifaceCall.ReturnArguments, ret)
+				}
+				ifaceCall.Once()
+			}
+			if tc.onRetArgsCmdList != nil {
+				mockCall := mockCmd.On(tc.onRetArgsCmdList.OnCallMethodName)
+				for _, arg := range tc.onRetArgsCmdList.OnCallMethodArgType {
+					mockCall.Arguments = append(mockCall.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range tc.onRetArgsCmdList.RetArgList {
+					mockCall.ReturnArguments = append(mockCall.ReturnArguments, ret)
+				}
+				mockCall.Once()
+			}
+
+			runner = tc.runnerInstance
+
+			e := ovsSet("table", "record")
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			}
+
+			mockCmd.AssertExpectations(t)
+			mockKexecIface.AssertExpectations(t)
+		})
+	}
+}
+
+func TestOvsFind(t *testing.T) {
+	mockKexecIface := new(mock_k8s_io_utils_exec.Interface)
+	mockCmd := new(mock_k8s_io_utils_exec.Cmd)
+
+	tests := []struct {
+		desc                string
+		expectedErr         error
+		onRetArgsKexecIface *ovntest.TestifyMockHelper
+		onRetArgsCmdList    *ovntest.TestifyMockHelper
+		runnerInstance      kexec.Interface
+	}{
+		{
+			desc:                "Test codepath when ovsExec returns an error",
+			expectedErr:         nil,
```

Comment:
per the test description, there is an indication that error is returned, but on L295 the expectedErr is set to nil, do you mind clariflying? Thanks